pydocstringformatter/formatting/base.py

```diff
@@ -35,14 +36,24 @@ def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
 class StringFormatter(Formatter):
     """Base class for formatter that only modifies the string content."""
 
+    quotes_regex = re.compile(r"""['"]{1,3}""")
+    """Pattern to match against opening quotes."""
+
     @abc.abstractmethod
-    def _treat_string(self, tokeninfo: tokenize.TokenInfo, indent_length: int) -> str:
+    def _treat_string(
+        self, tokeninfo: tokenize.TokenInfo, indent_length: int, quotes: str
+    ) -> str:
         """Return a modified string."""
 
     def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
+        # Get the quotes used for this docstring
+        match = re.match(self.quotes_regex, tokeninfo.string)
+        assert match
+        quotes = match.group()
+
         return tokenize.TokenInfo(
             tokeninfo.type,
-            self._treat_string(tokeninfo, tokeninfo.start[1]),
+            self._treat_string(tokeninfo, tokeninfo.start[1], quotes),
```

Comment:
> Maybe we could handle the quote detection in this function and give the string to the function ?

We're doing that now right?

> And maybe create a `StringWithOriginalQuoteFormatter` ?  It would make less arguments passed around ànd named `_` for `StringFormatter`

Good idea! I will do that!